tests/integration/service_test.py

```diff
@@ -813,6 +813,12 @@ def test_resolve_env(self):
         for k, v in {'FILE_DEF': 'F1', 'FILE_DEF_EMPTY': '', 'ENV_DEF': 'E3', 'NO_DEF': ''}.items():
             self.assertEqual(env[k], v)
 
+    def test_with_high_enough_api_version_we_get_default_network_mode(self):
+        with mock.patch.object(self.client, '_version', '1.20'):
+            service = self.create_service('web')
+            service_config = service._get_container_host_config({})
+            self.assertEquals(service_config['NetworkMode'], 'default')
```

Comment:
Hmm, that's weird, because that was the failure I was getting with docker 1.8 until @aanand helped me change the way it was written and patch the version number on the client object.